### ISSUE TITLE
[CoE Starter Kit - BUG] Orphaned Desktop Flows are inventoried every day to Dataverse Table Desktop Flows

### ISSUE THREAD

**Issue by DorotheevanTriel1** [microsoft/coe-starter-kit#4954]
### Describe the issue

After implementing the CoE Starter Kit Core Components for a client I was surprised that the number of Desktop Flows was increasing noticeably in a short period of time. 

I appears that orphaned Desktop Flows are inventoried every day to the Dataverse Table Desktop Flows. So every time the flow Admin | Sync Template v3 (Desktop flows) runs, a new entry for the exact same (orphaned) Desktop Flow is created, leading to an infinite number of records for one single flow. 

Deleting these Desktop Flows might be an option, however since there is no cleanup flow for Desktop Flows it will not result in a proper cleanup of the Desktop Flow data displayed in the CoE Starter Kit. 

### Expected Behavior

Every single Desktop Flow should be displayed only once in the corresponding Dataverse Table, even when the owner can not be resolved through AAD.

### What solution are you experiencing the issue with?

Core

### What solution version are you using?

4.30

### What app or flow are you having the issue with?

Admin | Sync Template v3 (Desktop flows)

**Comment by JeneferM-MSFT** [comment]
I've never seen this occur so can you please let me know

1. was this a clean install of Core 4.3 or an upgrade?
2. can you share a screenshot of these duplicate desktop flows in the the Power Platform Admin View? That will help me see what is duplicated and what isnt.

Thank you

**Comment by DorotheevanTriel1** [comment]
@JeneferM-MSFT @Jenefer-Monroe Any news on this topic? From my point of view this is a bug, not a question.

**Comment by Jenefer-Monroe** [comment]
We mark things as questions until they are understood. Its like the "seen and triaged" stage. 
We have not been able to investigate yet unfortunately. We have been preparing a preview release of the new architecture to present in [office hours](https://aka.ms/coeofficehours) tomorrow.

**Comment by Jenefer-Monroe** [comment]
OK I figured this one out. I've local fixed for next release (April). 

Here is how to local fix yourself. 
Open the **_Admin | Sync Template v3 (Desktop flows)_** flow
Go to Put Desktop Flows in CoE > For each Desktop Flow > Check that client data not equal to null > Update or Insert Desktop flow and Owner > (no branch) > Check if Orphan - New > (yes branch) > Update Record GUID as String - orphan
You will see that we wrote the field **_RPA_** into the wrong column :
![image](https://user-images.githubusercontent.com/89584709/225013450-bf9a8b62-6489-4303-89e9-c230e94fdd09.png)

Please cut from there and paste above into the correct, **_Record GUID as String_**, column.
It should now match the similar **_Update Record GUID as String_** action right next to it, as shown here
![image](https://user-images.githubusercontent.com/89584709/225014900-da137738-5adb-46ee-aff4-05384bec3b38.png)

**Comment by DorotheevanTriel1** [comment]
Thank you @JeneferM-MSFT @Jenefer-Monroe 

Is there any chance to clean up the thousands of duplicate entries for orphaned Desktop Flows that have been created in the Dataverse Table of my CoE environment automatically? Or do I have to delete the records from the Dataverse Table "Desktop Flows" manually?

**Comment by Jenefer-Monroe** [comment]
I'm working on that now, for some reason the '_' doesnt seem to work with the filters. So I'm seeing what I can do.

**Comment by Jenefer-Monroe** [comment]
ok I have **_CLEANUP - Admin | Sync Template v3 (Delete Bad Data)_** deleting the bad records for April release. Its a little complex so I just made a flow to do it for you that does the same thing as this helper. 

You'll import this flow to your CoE environment through this ux at http://flow.microsoft.com/ 
[deleteduplicatedesktopflowrecords_20230314152921.zip](https://github.com/microsoft/coe-starter-kit/files/10970515/deleteduplicatedesktopflowrecords_20230314152921.zip)

1. browse to  http://flow.microsoft.com/ > My flows > Import > Legacy Package
![image](https://user-images.githubusercontent.com/89584709/225053171-c0f2d0e9-e685-4e73-aab5-aae49c70ebcf.png)
![image](https://user-images.githubusercontent.com/89584709/225053866-b1a4aa30-b73c-47b9-998f-234217daf0b7.png)
2. Select the file and upload
![image](https://user-images.githubusercontent.com/89584709/225054041-a0846938-89ce-4bac-8ffe-c8dfdc27e828.png)
3. Select or create the connection and save
![image](https://user-images.githubusercontent.com/89584709/225054103-956d3201-b9c0-468c-8635-25bd70226679.png)
![image](https://user-images.githubusercontent.com/89584709/225054269-9c59be1a-8c1a-4be7-955b-a545f2baab2b.png)
4. Import
![image](https://user-images.githubusercontent.com/89584709/225054440-e0548378-344b-4a43-a266-4104c0e4897c.png)
5. Open the flow
![image](https://user-images.githubusercontent.com/89584709/225055132-61853850-43b5-4c1c-90ef-e2cfa060256b.png)
6. Save, then go back to turn on and run
![image](https://user-images.githubusercontent.com/89584709/225055432-f4891dd9-704f-48c1-98b1-c63947e05937.png)
![image](https://user-images.githubusercontent.com/89584709/225055578-e9b8ff22-a1cb-4bd2-8f8e-9b3f76dd303e.png)

**Comment by DorotheevanTriel1** [comment]
@Jenefer-Monroe Thank you! This was really helpful.

**Comment by CoEStarterKitBot** [comment]
@DorotheevanTriel1 This has been fixed in the latest release. Please install [the latest version of the toolkit](https://aka.ms/CoEStarterKitDownload) following the instructions for [installing updates](https://docs.microsoft.com/power-platform/guidance/coe/setup#installing-updates). Note that if you do not remove the unmanaged layers as described there you will not receive updates from us.
